modules/gateway/gateway.go

```diff
@@ -1,5 +1,115 @@
+// Package gateway connects a Sia node to the Sia flood network. The flood
+// network is used to propagate blocks and transactions. The gateway is the
+// primary avenue that a node uses to hear about transactions and blocks, and
+// is the primary avenue used to tell the network about blocks that you have
+// mined or about transactions that you have created.
 package gateway
 
+// For the user to be securely connected to the network, the user must be
+// connected to at least one node which will send them all of the blocks. An
+// attacker can trick the user into thinking that a different blockchain is the
+// full blockchain if the user is not connected to any nodes who are seeing +
+// broadcasting the real chain (and instead is connected only to attacker nodes
+// or to nodes that are not broadcasting). This situation is called an eclipse
+// attack.
+//
+// Connecting to a large number of nodes increases the resiliancy of the
+// network, but also puts a networking burden on the nodes and can slow down
+// block propagation or increase orphan rates. The gateway's job is to keep the
+// network efficient while also protecting the user against attacks.
+//
+// The gateway keeps a list of nodes that it knows about. It uses this list to
+// form connections with other nodes, and then uses those connections to
+// participate in the flood network. The primary vector for an attacker to
+// achieve an eclipse attack is node list domination. If a gateway's nodelist
+// is heavily dominated by attacking nodes, then when the gateway chooses to
+// make random connections the gateway is at risk of selecting only attacker
+// nodes.
+//
+// The gateway defends itself from these attacks by minimizing the amount of
+// control that an attacker has over the node list and peer list. The first
+// major defense is that the gateway maintais 8 'outbound' relationships, which
+// means that the gateway created those relationships instead of an attacker.
+// If a node forms a connection to you, that node is called 'inbound', and
+// because it may be an attacker node, it is not trusted. Outbound nodes can
+// also be attacker nodes, but they are less likely to be attacker nodes
+// because you chose them, instead of them choosing you.
+//
+// If the gateway forms too many connections, the gateway will allow incoming
+// connections by kicking an existing peer. But, to limit the amount of control
+// that an attacker may have, only inbound peers are selected to be kicked.
+// Furthermore, to increase the difficulty of attack, if a new inbound
+// connection shares the same IP address as an existing connection, the shared
+// connection is the connection that gets dropped (unless that connection is an
+// inbound connection).
```

Comment:
is "inbound" correct here? Seems like you might have meant "local" or "outbound" instead